apps/hpc-ftsadmin/src/app/components/reporting-detail.tsx

```diff
@@ -0,0 +1,498 @@
+            )}
+            options={verifiedOptions()}
+            controlledField={{
+              value: verified,
+              onChange: (value) => handleChange('verified', value),
+            }}
+            disabled={disabled}
+          />
+          <C.DatePicker
+            name="dateReported"
+            label={t.t(
+              lang,
+              (s) => s.components.reportingDetail.dateReported.label
+            )}
+            controlledField={{
+              value: dateReported,
+              onChange: (value) => handleChange('dateReported', value),
+            }}
+            disabled={disabled}
+            lang={lang}
+          />
+          <C.TextFieldWrapper
+            name="reporterReferenceCode"
+            label={t.t(
+              lang,
+              (s) => s.components.reportingDetail.reporterReferenceCode.label
+            )}
+            controlledField={{
+              value: reporterReferenceCode,
+              onChange: (value) => handleChange('reporterReferenceCode', value),
+            }}
+            disabled={disabled}
+          />
+          <C.TextFieldWrapper
+            name="reporterContactInfo"
+            label={t.t(
+              lang,
+              (s) => s.components.reportingDetail.reporterContactInfo.label
+            )}
+            textarea
+            minRows={2}
+            controlledField={{
+              value: reporterContactInfo,
+              onChange: (value) => handleChange('reporterContactInfo', value),
+            }}
+            disabled={disabled}
+          />
+        </div>
+      </Box>
+      <Box
+        sx={tw`flex gap-x-10 justify-around p-4 my-4 border border-solid border-unocha-panel-border rounded-sm flex-grow-0`}
+      >
+        <Box sx={tw`basis-1/2 max-w-[50%]`}>
+          <span>
+            {t.t(lang, (s) => s.components.reportingDetail.fileSection)}
+          </span>
+          <C.TextFieldWrapper
+            name="reportFileTitle"
+            label={t.t(
+              lang,
+              (s) => s.components.reportingDetail.reportFileTitle.label
+            )}
+            placeholder={t.t(
+              lang,
+              (s) => s.components.reportingDetail.reportFileTitle.label
+            )}
+            controlledField={{
+              value: reportFileTitle,
+              onChange: (value) => {
+                handleChange('reportFileTitle', value);
+              },
+            }}
+            disabled={disabled}
+          />
+
+          <C.UploadFile
+            name="file"
+            buttonConfig={{
+              color: 'primary',
+              text: t.t(lang, (s) => s.components.upload.buttonText),
+              startIcon: MdUploadFile,
+            }}
+            onUpload={handleUploadFile}
+            onDelete={handleDeleteFile}
+            onDownload={{
+              handleDownload: handleDownloadFile,
+              handleErrorToast: (error) => {
```

Comment:
The change where you add custom error toast when reporting file cannot be downloaded does not belong in the same commit where you "Rename map-functions to utils and add url validation". Separate this custom error work in another commit.